### PR TITLE
Add iis_site resource

### PR DESCRIPTION
**Pull request by chrisevett**
I modeled this after serverspec and then added a check for bindings. The janky format we check for (you'll notice some string mangling) matches the output that gets generated by the get-website cmdlet in powershell. I thought that would be the easiest way to write tests.

### CODE REVIEW

File: lib/resources/iis_site.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      end
+
+      bindings_array = []
+
+      # chop up our binding to look like it does when you run the get-website ps cmdlet
+      # using the following format
+      # <protocol> <bindingInformation> {if https "sslFlags=<sslFlags>"}
+      # the binding is a collection and there will always be at least one entry per site
+      if site['bindings']['Collection']
+        site['bindings']['Collection'].each do |binding|
```
Comment (chris-rock) [inline lib/resources/iis_site.rb]: It may be better to use http://ruby-doc.org/core-2.2.0/Array.html#method-i-map for that use case

**Comment by chris-rock** [comment]
@chrisevett Great effort, this is a great resource. Let just make some improvements. Compared to the original http://serverspec.org/resource_types.html#iis_website it is easier to read!
- [x] add documentation to https://github.com/chef/inspec/blob/master/docs/resources.rst and mark the resource as experimental
- [x] Steps to verify. Could you share a chef cookbook or powershell steps to get a sample IIS up and running with you tests? Normally we have integration tests for that: https://github.com/chef/inspec/tree/master/test/integration/default and https://github.com/chef/inspec#integration-tests

Let me know if something is unclear.

File: lib/resources/iis_site.rb (hunk 1 above)
Comment (chrisevett) [inline lib/resources/iis_site.rb]: Good suggestion. Thanks.

**Comment by chrisevett** [comment]
@chris-rock I'll make the updates you suggested. Thanks for the feedback.

File: lib/resources/iis_site.rb (hunk 1 above)
Comment (chrisevett) [inline lib/resources/iis_site.rb]: @chris-rock was something like this what you meant?

```
 site['bindings']['Collection'].map { |k, str|  
                                        k['protocol'] << 
                                        " " << 
                                        k['bindingInformation'] <<
                                        (k['protocol'] == 'https' ? ' sslFlags=' << flags : '')
                                      }
```

Comment (chris-rock) [inline lib/resources/iis_site.rb]: yeah!

Comment (chris-rock) [inline lib/resources/iis_site.rb]: The idea was to remove the `bindings_array = []` object

**Comment by chris-rock** [comment]
Hey @chrisevett Please let me know when you think its ready, I am going to review it then again

File: test/cookbooks/os_prepare/metadata.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@
 description 'This cookbook prepares the test operating systems'
 version '1.0.0'
 
-depends 'apt'
+depends 'iis'
```
Comment (chris-rock) [inline test/cookbooks/os_prepare/metadata.rb]: is there any reason you removed the apt cookbook?

**Comment by chrisevett** [comment]
@chris-rock 
This is good to go. In the interest of cutting down the clutter would it be better to smoosh this into one commit or would you rather merge it as is so we keep the history?

Thanks for being patient with my poor ruby skills.

**Comment by chris-rock** [comment]
@chrisevett it would be great if you could squash that to no more that 5 commits. `git rebase -i master` is a good tool for that.

File: test/cookbooks/os_prepare/recipes/iis_site.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# set up test site for iis_site resource 
+
+return unless node['platform_family'] == 'windows'
+  
+  # make sure the iis windows feature is installed
```
Comment (chris-rock) [inline test/cookbooks/os_prepare/recipes/iis_site.rb]: I think we do not need the indent here.

File: test/cookbooks/os_prepare/recipes/iis_site.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  dsc_script 'Web-Server' do
```
Comment (chris-rock) [inline test/cookbooks/os_prepare/recipes/iis_site.rb]: I assume we do not need the iis cookbook then in our depends list. Right?

File: test/cookbooks/os_prepare/metadata.rb (hunk 2 above)
Comment (chrisevett) [inline test/cookbooks/os_prepare/metadata.rb]: I undid this change.

File: test/cookbooks/os_prepare/recipes/iis_site.rb (hunk 3 above)
Comment (chrisevett) [inline test/cookbooks/os_prepare/recipes/iis_site.rb]: I removed the indent.

File: test/cookbooks/os_prepare/recipes/iis_site.rb (hunk 4 above)
Comment (chrisevett) [inline test/cookbooks/os_prepare/recipes/iis_site.rb]: Correct. Thanks!

**Comment by chrisevett** [comment]
@chris-rock this is ready for review once more. Thanks!

File: docs/resources.rst [hunk 5]
```diff
@@ -1799,7 +1800,111 @@ The following examples show how to use this InSpec audit resource.
...
+  it { should have_app_pool('app_pool') }
+  it { should have_binding('https :443:www.contoso.com sslFlags=0') }
+  it { should have_path('C:\\inetpub\\wwwroot') }
+end
+
+where
+
+* ``iis_site()`` must specify a web site name
+* ``'website'`` is the web site name
+* ``'service_name'`` is a service listed in the ``inetd.conf`` file
```
Comment (chris-rock) [inline docs/resources.rst]: I think we do not need the `service_name` here

File: docs/resources.rst [hunk 6]
```diff
@@ -1799,7 +1800,111 @@ The following examples show how to use this InSpec audit resource.
...
+* ``have_app_pool('my_app_pool')`` tests that our site belongs to the 'my_app_pool' application pool
+* ``have_binding('my_binding')`` tests that our site has the specified binding. my_binding should be in the format of the default output from the Get-Website powershell cmdlet
+
+website
```
Comment (chris-rock) [inline docs/resources.rst]: Lets remove `website`

File: test/integration/default/iis_site_spec.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  its(:stdout) {should match '.*?Default Web Site.*?'}
+end
+
+# test the site with the resource 
+describe iis_site('Default Web Site') do
+  it { should exist }
+  it { should be_running }
+  it { should have_app_pool('DefaultAppPool') }
+  it { should have_binding('http *:80:') }
+  it { should have_path('c:\\inetpub\\wwwroot\\') }
```
Comment (chris-rock) [inline test/integration/default/iis_site_spec.rb]: I'll receive `%SystemDrive%\\inetpub\\wwwroot` as a result.

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chris-rock** [comment]
Awesome work @chrisevett The resource works great 👍 I'll added some super minor improvements. Then lets just rebase this PR to latest master via `git rebase master`

**Comment by chris-rock** [comment]
I'll merge this PR as soon as you're done

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by chrisevett** [comment]
Rebase and edits are complete. Thanks.

**Comment by chris-rock** [comment]
Thanks again for hard work @chrisevett 💯

**Pull request merged**